pyccel/ast/core.py

```diff
@@ -429,11 +436,15 @@ def __init__(self, variable, *, shape, status, like = None):
         if status not in ('allocated', 'unallocated', 'unknown'):
             raise ValueError(f"Value of 'status' not allowed: '{status}'")
 
+        if alloc_type:
+            assert alloc_type in ('init', 'reserve', 'resize')
```

Comment:
```suggestion
        assert alloc_type is None or alloc_type in ('init', 'reserve', 'resize')
```

Reply:
Should there also be an assertion e.g.:
```python
assert alloc_type is None || not isinstance(variable.class_type, (HomogeneousListType, HomogeneousSetType, DictType))
```
?

Reply:
Yes! this is a very good. I will add it!

Reply:
fixed